Src/ElasticScale.Client/Query/MultiShardDataReader.cs

```diff
@@ -1062,7 +1065,9 @@ public override bool Read()
                 //
                 if (this.PerformReadToFillBuffer())
                 {
-                    s_tracer.Verbose("MultiShardDataReader.Read.Complete; Duration: {0}; Shard: {1}", DateTime.UtcNow - readStartTime,
+                    stopwatch.Stop();
```

Comment:
Should stopwatch.Stop() also be called for the early return false above at line 1061?